system-functionalities.md

```diff
@@ -208,18 +218,17 @@ Protocol:
         1. The `user_id` must be of the expected format and length, and should match that of the open request session,
         1. The `key_id` must be associated with the given `user_id` in the key server's database.
         1. The intended use must match one of the expected options.
-    1. [Retrieves](#server-side-storage) the associated ciphertext, `ciphertext` and associated data, `associated_data`, for the given pair `(user_id, key_id)` from its database and sends `ciphertext` and `associated data` to the client.
+    1. [Retrieves](#server-side-storage) the record tuple `record` for the given pair `(user_id, key_id)` from its database, updates the database record to indicate that the secret has been retrieved by the client, and sends `record` to the client.
```

Comment:
Do you mean there will be a permanent line for each secret in the database that switches from "never retrieved" to "retrieved" the first time we do this? Would it be recorded the same way if the key was imported? 

This sounds okay to me, but I want to avoid it turning into a log (e.g. listing out time + date for every retrieval).

Reply:
Yes, a permanent flag of some kind. I don't know if we should implement this in the same way as for imported keys or not. I think this captures different information. For imported keys, I suspect we want them to be a wrapper around a key type, so that the object itself tells us they are imported. That said, I already included context as part of the key itself to track if the key is imported, I believe.

Reply:
Okay, cool. I was thinking a single flag for "this key has existed outside the system" but I don't know that it's worth / useful to simplifying that.